src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.cs

```diff
@@ -88,6 +89,12 @@ private void SetHttpContext(HttpContext httpContext)
             formData.SetFormData(handler, new FormCollectionReadOnlyDictionary(form));
         }
 
+        var antiforgery = httpContext.RequestServices.GetRequiredService<AntiforgeryStateProvider>();
+        if (antiforgery is EndpointAntiforgeryStateProvider endpointAntiforgery)
+        {
+            endpointAntiforgery.SetRequestContext(httpContext);
+        }
```

Comment:
If we're willing to skip this if the service isn't `EndpointAntiforgeryStateProvider`, could we treat the service as optional and use `GetService` instead of `GetRequiredService`? Might simplify some testing scenarios.

Reply:
done